ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -542,14 +542,18 @@ default boolean isTimeTravelAllowed() {
     return false;
   }
 
-  default boolean isMetadataTableSupported() {
+  default boolean isTableIdentifierSupported() {
     return false;
   }
 
   default boolean isValidMetadataTable(String metaTableName) {
     return false;
   }
 
+  default boolean isValidBranch(org.apache.hadoop.hive.metastore.api.Table hmsTable, String branchName) {
```

Comment:
other storageHandlers might not have branch support at all

Reply:
I agree you. But here i want to check the branch validity in `compile ` stage instead of iceberg `commit `stage , and i didn't find good way except using  `HiveStorageHandler` to invoke iceberg api  in `ql `module.

Reply:
I removed this method `isValidBranch` and added a new method `HiveStorageHandler::checkAndSetTableMetaRef` which can check and set tableMetaRef(metadata, branch, tag), and this can make  `Hive::getTable` of `ql ` module more clean as the tableMetaRef is specific to iceberg.

Reply:
👍